torch_xla/csrc/tensor.cpp

```diff
@@ -1206,23 +1212,33 @@ std::vector<XLATensor> XLATensor::split(const XLATensor& input,
                                         xla::int64 split_size, xla::int64 dim) {
   auto input_shape = input.shape();
   int split_dim = GetCanonicalDimensionIndex(dim, input_shape.get().rank());
-  xla::int64 size_in_dim = input_shape.get().dimensions(split_dim);
-  // Deal with 0 split size, it's a corner case which is only allowed when the
-  // dimension size is 0 as well.
+  xla::int64 dim_size = input_shape.get().dimensions(split_dim);
   if (split_size == 0) {
-    XLA_CHECK_EQ(size_in_dim, 0);
+    // Deal with 0 split size, it's a corner case which is only allowed when the
+    // dimension size is 0 as well.
+    XLA_CHECK_EQ(dim_size, 0);
     xla::Literal literal(input_shape.get());
     return {Create(ir::MakeNode<ir::ops::Constant>(std::move(literal)),
                    input.GetDevice())};
   }
-  xla::int64 chunks = xla::CeilOfRatio(size_in_dim, split_size);
-  ir::NodePtr node =
-      ir::MakeNode<ir::ops::Split>(input.GetIrValue(), split_size, split_dim);
-  std::vector<XLATensor> result;
-  for (xla::int64 i = 0; i < chunks; ++i) {
-    result.push_back(Create(ir::Value(node, i), input.GetDevice()));
+  std::vector<xla::int64> split_sizes;
+  for (; dim_size > 0; dim_size -= split_size) {
```

Comment:
I'd use a `while` here, it's almost as concise and more readable.

Reply:
Why?
I can understand "for (; COND;)" is silly.
I can even (barely) understand a "for(;;)" -> "while (true)" migration.
But "for (; COND; POST)" expanded to a while it is objectively more concise, and personally more readable ;)